src/Microsoft.DotNet.Wpf/src/PresentationFramework/Microsoft/Win32/CommonItemDialog.cs

```diff
@@ -0,0 +1,693 @@
+            }
+        }
+
+        private protected string[] MutableFileNames
+        {
+            get { return _fileNames; }
+            set { _fileNames = value; }
+        }
+
+        /// <summary>
+        ///  In cases where we need to return an array of strings, we return
+        ///  a clone of the array.  We also need to make sure we return a 
+        ///  string[0] instead of a null if we don't have any filenames.
+        /// </summary>
+        private protected string[] CloneFileNames()
+        {
+            if (_fileNames == null)
+            {
+                return Array.Empty<string>();
+            }
+            else
+            {
+                return (string[])_fileNames.Clone();
+            }
+        }
+
+        #endregion Internal Properties
+
+        //---------------------------------------------------
+        //
+        // Internal Events
+        //
+        //---------------------------------------------------
+        //#region Internal Events
+        //#endregion Internal Events
+
+        //---------------------------------------------------
+        //
+        // Private Methods
+        //
+        //---------------------------------------------------
+        #region Private Methods
+
+        //  Provides the actual implementation of initialization tasks.
+        //  Initialize() is called from both the constructor and the
+        //  public Reset() function to set default values for member
+        //  variables and for the options bitmask.
+        private void Initialize()
+        {
+            // 
+            // Initialize Options Flags
+            // 
+            _dialogOptions.Value = 0;   // _dialogOptions is an int containing a set of
+                                        // bit flags used to initialize the dialog box.
+                                        // Within our code, we only use GetOption and SetOption
+                                        // (change from Windows Forms, which sometimes directly
+                                        // modified _dialogOptions).  As such, we initialize to 0
+                                        // here and then call SetOption to get _dialogOptions
+                                        // into the default state.
+
+            //
+            // Set some default options
+            //
+            // - Specifies that the user can type only valid paths and file names. If this flag is
+            //   used and the user types an invalid path and file name in the File Name entry field,
+            //   we will display a warning in a message box.
+            SetOption(FOS.PATHMUSTEXIST, true);
+
+            // - Force no mini mode for the SaveFileDialog.
+            SetOption(FOS.DEFAULTNOMINIMODE, true);
+
+            //
+            // Initialize additional properties
+            // 
+            _fileNames = null;
+            _title.Value = null;
+            _initialDirectory.Value = null;
+
```

Comment:
We should set `ClientGuid = null` here as well.

Reply:
Good catch! Fixed in https://github.com/dotnet/wpf/pull/7916/commits/38ab72d13467b15166947fb783a5cb7ae4bd0079